transparentdemocracy/documents/summarize.py

```diff
@@ -18,72 +22,33 @@
 logger = logging.getLogger("__name__")
 logger.setLevel(logging.INFO)
 
+# Batch size doesn't really matter when using OLLAMA locally, it gets executed sequentially anyway
+BATCH_SIZE = 1
 SUMMARY_DOCUMENT_FILENAME_PATTERN = re.compile("^.*/55K(\\d{4})(\\d{3}).summary$")
 
-# Config for llama3
-OLLAMA_MODEL = "llama3"  # 7b parameters
-# must match model, see https://en.wikipedia.org/wiki/Llama_(language_model)
-CONTEXT_WINDOW = 8912
-
-#
-SUMMARY_SIZE = 400  # number of tokens for each summary
-# A word is +- 1.3 tokens
-WORDS_PER_DOCUMENT = int((CONTEXT_WINDOW - SUMMARY_SIZE) / 1.3)
-
-PROMPT_BRIEFNESS_NL = "Introduceer het antwoord niet, maar schrijf enkel de samenvatting"
-PROMPT_VOCAB_NL = "Gebruik woordenschat die geschikt is voor leken"
-
-PROMPT_BRIEFNESS_FR = "N'introduisez pas la réponse, rédigez simplement le résumé"
-PROMPT_VOCAB_FR = "Utilisez un vocabulaire adapté aux novices"
-
-PROMPT_STUFF_NL = f""""Vat de volgende tekst samen in het Nederlands. {PROMPT_BRIEFNESS_NL}. {PROMPT_VOCAB_NL}.
-                 {{text}}
-                 BONDIGE SAMENVATTING:"""
-PROMPT_MAP_NL = f"""Vat de volgende tekst samen in het Nederlands. {PROMPT_BRIEFNESS_NL}.
-                 {{text}}
-                 BONDIGE SAMENVATTING:"""
-PROMPT_REDUCE_NL = f"""Hierna volgen enkele samenvattingen. Maak een geconsolideerde samenvatting in het Nederlands.  {PROMPT_BRIEFNESS_NL}. {PROMPT_VOCAB_NL}.
-                {{text}}
-                BONDIGE SAMENVATTING:"""
-
-PROMPT_STUFF_FR = f""""Résumez le texte suivant en français. {PROMPT_BRIEFNESS_FR}. {PROMPT_VOCAB_FR}.
-                 {{text}}
-                 Résumé concis:"""
-PROMPT_MAP_FR = f"""Résumez le texte suivant en français. {PROMPT_BRIEFNESS_FR}.
-                 {{text}}
-                 Résumé concis:"""
-PROMPT_REDUCE_FR = f"""Ci-dessous quelques résumés. Créez un résumé consolidé en français.  {PROMPT_BRIEFNESS_FR}. {PROMPT_VOCAB_FR}.
-                {{text}}
-                Résumé concis:"""
-
-PROMPTS = dict(
-    nl=(PROMPT_STUFF_NL, PROMPT_MAP_NL, PROMPT_REDUCE_NL),
-    fr=(PROMPT_STUFF_FR, PROMPT_MAP_FR, PROMPT_REDUCE_FR)
-)
-
-
-class DocumentSummarizer():
-    def __init__(self, language="nl"):
+OLLAMA_MODEL = "llama3"
+PROMPT_STUFF = """Summarize the text in Dutch and in French. Here is the text:
+
+{text}
+
+Answer with a single json object. The dutch summary should be in a property called nl and the french summary should be in a property called fr."""
```

Comment:
Does this consistently work?
I've found in json outputting instructions for an LLM at work, the LLM sometimes does not return proper JSON if you write out in "vage human terms" what to generate.
If so: we make it very explicit, like:

```
Return a JSON object, containing a Dutch summary (let's refer to this with the variable [DUTCH_SUMMARY]) and French summary (let's refer to this as [FRENCH_SUMMARY]) of the following text:

{text}

The returned JSON object must look like this:

{
  "nl": "[DUTCH_SUMMARY]",
  "fr": "[FRENCH_SUMMARY]"
}

The summary must not be longer than 5 sentences.
It should be written in layman terms, rather than using very judicial or political vocabulary.
```

And even an in-context example if you still find the LLM makes mistakes sometimes:

```
For example, this text:

Dit wetsontwerp reguleert de privé-opsporing van minderjarigen en autoriseert specifiek een privédetective om met toestemming van de ouders een kind jonger dan 12 jaar te ondervragen. De auteur vindt dat dit te ver gaat en kan leiden tot instrumentalisering door ouders. Daarom wordt voorgesteld deze bevoegdheid weg te laten en het getuigenis van minderjarigen alleen mogelijk te maken binnen het kader van een gerechtelijke procedure met alle daaraan verbonden garanties.

Must result in a JSON of this structure:

{
  "nl": "Laat toe om kinderen jonger dan 12 jaar te laten ondervragen door privé-detectives, indien de ouders van het kind toestemmen.",
  "fr": "Permettez une interrogation d'un enfant de moins de 12 ans par un détective privé, si les parents de l'enfant le permettent."
}
```